apps/aecore/src/aec_conductor.erl

```diff
@@ -380,10 +376,14 @@ handle_monitor_message(Ref, Pid, Why, State) ->
             State;
         {ok, Tag} ->
             epoch_mining:error("Worker died: ~p", [{Tag, Pid, Why}]),
-            State1 = State#state{workers = orddict:erase(Pid, Workers),
-                                 blocked_tags = ordsets:del_element(Tag, Blocked)
-                                },
-            start_mining(State1)
+            State1 = case Tag of
+                         mining -> deregister_miner_instance(State, Pid);
+                         _Other -> State
+                     end,
+            State2 = State1#state{workers      = orddict:erase(Pid, Workers),
+                                  blocked_tags = ordsets:del_element(Tag, Blocked)
```

Comment:
This is not necessary for miners, right?

Reply:
True, will make it dependent on worker type.